[
  {
    "url": "https://api.github.com/repos/simonw/playing-with-git-notes/issues/comments/1296302460",
    "html_url": "https://github.com/simonw/playing-with-git-notes/issues/1#issuecomment-1296302460",
    "issue_url": "https://api.github.com/repos/simonw/playing-with-git-notes/issues/1",
    "id": 1296302460,
    "node_id": "IC_kwDOIVsEAM5NRAF8",
    "user": {
      "login": "simonw",
      "id": 9599,
      "node_id": "MDQ6VXNlcjk1OTk=",
      "avatar_url": "https://avatars.githubusercontent.com/u/9599?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/simonw",
      "html_url": "https://github.com/simonw",
      "followers_url": "https://api.github.com/users/simonw/followers",
      "following_url": "https://api.github.com/users/simonw/following{/other_user}",
      "gists_url": "https://api.github.com/users/simonw/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/simonw/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/simonw/subscriptions",
      "organizations_url": "https://api.github.com/users/simonw/orgs",
      "repos_url": "https://api.github.com/users/simonw/repos",
      "events_url": "https://api.github.com/users/simonw/events{/privacy}",
      "received_events_url": "https://api.github.com/users/simonw/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2022-10-30T17:00:50Z",
    "updated_at": "2022-10-30T17:00:50Z",
    "author_association": "OWNER",
    "body": "    git notes add -m \"My first note\"\r\n\r\nThis seemed to attach a note to the current commit.\r\n\r\n    git notes add -m \"My first note\" main\r\n\r\nThat had the same effect - actually it complained that a note already existed, so I had to add `-f` to replace it.\r\n\r\n    git notes add -m \"Added to specific commit 5a01208a441ff9109c894259decbcd4b8aba1681\" 5a01208a441ff9109c894259decbcd4b8aba1681 -f\r\n\r\nProblem is seeing the notes! Running `git notes` returns a not very useful list:\r\n```\r\ngit notes\r\n814c49c5c98ae6083a48af9dc8487f8ad98aebce 5a01208a441ff9109c894259decbcd4b8aba1681\r\n3a8bf47ed13c85db8298bff9781721fc55814d70 e7294b87c2e939897d195122d27b1e6b0302176e\r\n```\r\nBut `git log` locally does the right thing:\r\n```\r\ncommit e7294b87c2e939897d195122d27b1e6b0302176e (HEAD -> main, origin/main)\r\nAuthor: Simon Willison <swillison@gmail.com>\r\nDate:   Sun Oct 30 09:55:06 2022 -0700\r\n\r\n    Remove GitHub Actions workflows\r\n\r\nNotes:\r\n    A note against current main\r\n\r\ncommit 5a01208a441ff9109c894259decbcd4b8aba1681\r\nAuthor: Simon Willison <swillison@gmail.com>\r\nDate:   Sun Oct 30 09:53:53 2022 -0700\r\n\r\n    Update README.md\r\n\r\nNotes:\r\n    Added to specific commit 5a01208a441ff9109c894259decbcd4b8aba1681\r\n```",
    "reactions": {
      "url": "https://api.github.com/repos/simonw/playing-with-git-notes/issues/comments/1296302460/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  },
  {
    "url": "https://api.github.com/repos/simonw/playing-with-git-notes/issues/comments/1296302643",
    "html_url": "https://github.com/simonw/playing-with-git-notes/issues/1#issuecomment-1296302643",
    "issue_url": "https://api.github.com/repos/simonw/playing-with-git-notes/issues/1",
    "id": 1296302643,
    "node_id": "IC_kwDOIVsEAM5NRAIz",
    "user": {
      "login": "simonw",
      "id": 9599,
      "node_id": "MDQ6VXNlcjk1OTk=",
      "avatar_url": "https://avatars.githubusercontent.com/u/9599?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/simonw",
      "html_url": "https://github.com/simonw",
      "followers_url": "https://api.github.com/users/simonw/followers",
      "following_url": "https://api.github.com/users/simonw/following{/other_user}",
      "gists_url": "https://api.github.com/users/simonw/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/simonw/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/simonw/subscriptions",
      "organizations_url": "https://api.github.com/users/simonw/orgs",
      "repos_url": "https://api.github.com/users/simonw/repos",
      "events_url": "https://api.github.com/users/simonw/events{/privacy}",
      "received_events_url": "https://api.github.com/users/simonw/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2022-10-30T17:01:49Z",
    "updated_at": "2022-10-30T17:01:49Z",
    "author_association": "OWNER",
    "body": "Running `git push` didn't seem to do anything with the notes.\r\n\r\nhttps://stackoverflow.com/questions/18268986/git-how-to-push-messages-added-by-git-notes-to-the-central-git-server suggested using the following:\r\n\r\n     git push origin 'refs/notes/*'\r\n\r\n(I added single quotes to get this working in `zsh`).\r\n\r\nThat published them... but they don't seem to show up anywhere in the GitHub interface.",
    "reactions": {
      "url": "https://api.github.com/repos/simonw/playing-with-git-notes/issues/comments/1296302643/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  },
  {
    "url": "https://api.github.com/repos/simonw/playing-with-git-notes/issues/comments/1296302978",
    "html_url": "https://github.com/simonw/playing-with-git-notes/issues/1#issuecomment-1296302978",
    "issue_url": "https://api.github.com/repos/simonw/playing-with-git-notes/issues/1",
    "id": 1296302978,
    "node_id": "IC_kwDOIVsEAM5NRAOC",
    "user": {
      "login": "simonw",
      "id": 9599,
      "node_id": "MDQ6VXNlcjk1OTk=",
      "avatar_url": "https://avatars.githubusercontent.com/u/9599?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/simonw",
      "html_url": "https://github.com/simonw",
      "followers_url": "https://api.github.com/users/simonw/followers",
      "following_url": "https://api.github.com/users/simonw/following{/other_user}",
      "gists_url": "https://api.github.com/users/simonw/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/simonw/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/simonw/subscriptions",
      "organizations_url": "https://api.github.com/users/simonw/orgs",
      "repos_url": "https://api.github.com/users/simonw/repos",
      "events_url": "https://api.github.com/users/simonw/events{/privacy}",
      "received_events_url": "https://api.github.com/users/simonw/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2022-10-30T17:03:40Z",
    "updated_at": "2022-10-30T17:03:40Z",
    "author_association": "OWNER",
    "body": "I tried a fresh checkout:\r\n\r\n```\r\ncd /tmp\r\ngit clone https://github.com/simonw/playing-with-git-notes\r\n```\r\nAnd `git log` did not show my notes.\r\n\r\nThen I ran:\r\n```\r\ngit fetch origin 'refs/notes/*:refs/notes/*'\r\n```\r\n```\r\nremote: Enumerating objects: 11, done.\r\nremote: Counting objects: 100% (11/11), done.\r\nremote: Compressing objects: 100% (7/7), done.\r\nremote: Total 11 (delta 2), reused 11 (delta 2), pack-reused 0\r\nUnpacking objects: 100% (11/11), 1.08 KiB | 138.00 KiB/s, done.\r\nFrom https://github.com/simonw/playing-with-git-notes\r\n * [new ref]         refs/notes/commits -> refs/notes/commits\r\n```\r\nAnd now `git log` in that repo shows my notes correctly.",
    "reactions": {
      "url": "https://api.github.com/repos/simonw/playing-with-git-notes/issues/comments/1296302978/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  },
  {
    "url": "https://api.github.com/repos/simonw/playing-with-git-notes/issues/comments/1296303072",
    "html_url": "https://github.com/simonw/playing-with-git-notes/issues/1#issuecomment-1296303072",
    "issue_url": "https://api.github.com/repos/simonw/playing-with-git-notes/issues/1",
    "id": 1296303072,
    "node_id": "IC_kwDOIVsEAM5NRAPg",
    "user": {
      "login": "simonw",
      "id": 9599,
      "node_id": "MDQ6VXNlcjk1OTk=",
      "avatar_url": "https://avatars.githubusercontent.com/u/9599?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/simonw",
      "html_url": "https://github.com/simonw",
      "followers_url": "https://api.github.com/users/simonw/followers",
      "following_url": "https://api.github.com/users/simonw/following{/other_user}",
      "gists_url": "https://api.github.com/users/simonw/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/simonw/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/simonw/subscriptions",
      "organizations_url": "https://api.github.com/users/simonw/orgs",
      "repos_url": "https://api.github.com/users/simonw/repos",
      "events_url": "https://api.github.com/users/simonw/events{/privacy}",
      "received_events_url": "https://api.github.com/users/simonw/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2022-10-30T17:04:13Z",
    "updated_at": "2022-10-30T17:04:13Z",
    "author_association": "OWNER",
    "body": "So it looks like GitHub's support for `git notes` is basically non-existent? You can push and pull notes explicitly if you know what you are doing but they don't seem to show up anywhere in the GitHub web UI.",
    "reactions": {
      "url": "https://api.github.com/repos/simonw/playing-with-git-notes/issues/comments/1296303072/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  },
  {
    "url": "https://api.github.com/repos/simonw/playing-with-git-notes/issues/comments/1296303913",
    "html_url": "https://github.com/simonw/playing-with-git-notes/issues/1#issuecomment-1296303913",
    "issue_url": "https://api.github.com/repos/simonw/playing-with-git-notes/issues/1",
    "id": 1296303913,
    "node_id": "IC_kwDOIVsEAM5NRAcp",
    "user": {
      "login": "simonw",
      "id": 9599,
      "node_id": "MDQ6VXNlcjk1OTk=",
      "avatar_url": "https://avatars.githubusercontent.com/u/9599?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/simonw",
      "html_url": "https://github.com/simonw",
      "followers_url": "https://api.github.com/users/simonw/followers",
      "following_url": "https://api.github.com/users/simonw/following{/other_user}",
      "gists_url": "https://api.github.com/users/simonw/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/simonw/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/simonw/subscriptions",
      "organizations_url": "https://api.github.com/users/simonw/orgs",
      "repos_url": "https://api.github.com/users/simonw/repos",
      "events_url": "https://api.github.com/users/simonw/events{/privacy}",
      "received_events_url": "https://api.github.com/users/simonw/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2022-10-30T17:07:49Z",
    "updated_at": "2022-10-30T17:07:49Z",
    "author_association": "OWNER",
    "body": "Started a Twitter thread: https://twitter.com/simonw/status/1586766260842266625",
    "reactions": {
      "url": "https://api.github.com/repos/simonw/playing-with-git-notes/issues/comments/1296303913/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  },
  {
    "url": "https://api.github.com/repos/simonw/playing-with-git-notes/issues/comments/1296304250",
    "html_url": "https://github.com/simonw/playing-with-git-notes/issues/1#issuecomment-1296304250",
    "issue_url": "https://api.github.com/repos/simonw/playing-with-git-notes/issues/1",
    "id": 1296304250,
    "node_id": "IC_kwDOIVsEAM5NRAh6",
    "user": {
      "login": "simonw",
      "id": 9599,
      "node_id": "MDQ6VXNlcjk1OTk=",
      "avatar_url": "https://avatars.githubusercontent.com/u/9599?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/simonw",
      "html_url": "https://github.com/simonw",
      "followers_url": "https://api.github.com/users/simonw/followers",
      "following_url": "https://api.github.com/users/simonw/following{/other_user}",
      "gists_url": "https://api.github.com/users/simonw/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/simonw/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/simonw/subscriptions",
      "organizations_url": "https://api.github.com/users/simonw/orgs",
      "repos_url": "https://api.github.com/users/simonw/repos",
      "events_url": "https://api.github.com/users/simonw/events{/privacy}",
      "received_events_url": "https://api.github.com/users/simonw/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2022-10-30T17:09:06Z",
    "updated_at": "2022-10-30T17:09:06Z",
    "author_association": "OWNER",
    "body": "Oh frustrating: https://github.blog/2010-08-25-git-notes-display/\r\n\r\n> Update (August 14, 2014): Displaying Git notes on GitHub is no longer supported.",
    "reactions": {
      "url": "https://api.github.com/repos/simonw/playing-with-git-notes/issues/comments/1296304250/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  }
]
